geode-core/src/main/java/org/apache/geode/pdx/internal/json/PdxToJSON.java

```diff
@@ -224,9 +226,15 @@ private void getJSONStringFromArray(JsonGenerator jg, Object value, String pf)
         writeValue(jg, obj, pf);
       }
       jg.writeEndArray();
+    } else if (value.getClass().isArray()) {
+      throw new IllegalStateException(
+          "The pdx field " + pf + " is an array whose component type "
+              + value.getClass().getComponentType()
+              + " can not be converted to JSON.");
     } else {
       throw new IllegalStateException(
-          "PdxInstance returns unknown pdxfield " + pf + " for type " + value);
+          "Expected an array for pdx field " + pf + ", but got an object of type "
```

Comment:
since only array object will call this method, this "else" statement will never be executed. I think this if/else is redundant.

Reply:
As in my reply to Darrel's similar earlier comment, I'm not sure I like trusting that the caller made sure that value.getClass().isArray() was true and there's nothing about the method signature (and I don't believe can be) that would prevent this, so my instinct is to go with "defensive coding". (What's "safe" and "obvious" today can get lost in future changes.)

Reply:
Yeah, that's a good point if it's a public api, but this method was intended to be "private". 
I am fine leaving it as is though.

Reply:
Thanks, I did go back and forth about it, since it is a private method, but finally settled on playing it safe since even within a class when the lines of code increases assumptions can get lost. I do agree that the isArrary check should come first and, since I was already going to be doing another commit, went ahead and made the change.